packages/components/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,202 @@
+                activeKnobAnimation.start();
+                activeBackgroundAnimation.start();
+            } else {
+                inactiveKnobAnimation.start();
+                inactiveBackgroundAnimation.start();
+            }
+        }, [active]);
+
+        const handlePress = () => {
+            const newState = !active;
+            onChange(newState);
+        };
+
+        return (
+            <PressableHighlight
+                disabled={disabled || loading}
+                onPress={handlePress}
+                style={styles.pressableContainer}
+            >
+                <View ref={ref} style={[styles.toggleContainer, rest.style]}>
+                    <View style={styles.statusBackground}></View>
+
+                    <Animated.View
+                        style={[
+                            styles.animatedBackground,
+                            {
+                                transform: [
+                                    {
+                                        translateX: backgroundProgressValue.interpolate({
+                                            inputRange: [0, 1],
+                                            outputRange: [-(WIDTH - KNOB_SIZE) / 2, 0],
+                                            extrapolate: "clamp",
+                                        }),
+                                    },
+                                    { scaleX: backgroundDisplacement },
+                                ],
+                            },
+                        ]}
+                    ></Animated.View>
+
+                    <Animated.View
+                        style={{ position: "absolute", transform: [{ translateX: displacement }] }}
+                    >
+                        <Paper elevation="raised" style={styles.knob} />
+                    </Animated.View>
+                </View>
+            </PressableHighlight>
+        );
+    },
+);
+
+Switch.displayName = "Switch";
+
+type ToggleStyleSheetProps = {
+    isActive: boolean;
+    color: "primary" | "secondary" | "danger";
+    disabled: boolean;
+};
+
+const themeStyles = EDSStyleSheet.create((theme, props: ToggleStyleSheetProps) => {
+    const { color, disabled, isActive } = props;
+
+    const activeBackgroundColor = theme.colors.interactive[color];
+    const inactiveBackgroundColor = theme.colors.interactive.disabled;
+
+    const knobColor = theme.colors.text.tertiary;
+    const disabledKnobColor = theme.colors.text.disabled;
+    const backgroundHeight = KNOB_SIZE * 0.5;
+
+    let backgroundColor = isActive ? activeBackgroundColor : inactiveBackgroundColor;
+    backgroundColor = disabled ? theme.colors.interactive.disabled : backgroundColor;
+
+    return {
+        toggleContainer: {
+            flexDirection: "row",
+            alignItems: "center",
+        },
+        pressableContainer: {
+            height: HEIGHT,
+            width: WIDTH,
+            padding: 7,
+            borderRadius: 100,
```

Comment:
Just to safe, maybe set the border radius to be WIDTH / 2 so that it scales with size